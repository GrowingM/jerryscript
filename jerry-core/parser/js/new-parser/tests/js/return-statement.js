/* This test contains throw statement parser tests. */

next_statement;

//function return_undefined() {
    return;
    { return }
//}

next_statement;

//function return_expression() {
    { return exp }
//}

next_statement;

//function return_undefined_and_dead_code() {
    return
    exp
//}

next_statement;
