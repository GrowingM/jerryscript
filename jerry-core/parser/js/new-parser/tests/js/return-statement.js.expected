 [  1] CBC_PUSH_IDENT ident-'next_statement'-id:0
 [  0] CBC_POP
 [  0] CBC_RETURN_WITH_UNDEFINED
 [  0] CBC_RETURN_WITH_UNDEFINED
 [  1] CBC_PUSH_IDENT ident-'next_statement'-id:0
 [  0] CBC_POP
 [  1] CBC_PUSH_IDENT ident-'exp'-id:1
 [  0] CBC_RETURN
 [  1] CBC_PUSH_IDENT ident-'next_statement'-id:0
 [  0] CBC_POP
 [  0] CBC_RETURN_WITH_UNDEFINED
 [  1] CBC_PUSH_IDENT ident-'exp'-id:1
 [  0] CBC_POP
 [  1] CBC_PUSH_IDENT ident-'next_statement'-id:0
 [  0] CBC_POP

Final byte code:
  Maximum stack depth: 1
  Literal encoding: byte
  Number of var literals: 0
  Number of identifiers: 2
  Number of other literals: 0

   0 : CBC_PUSH_IDENT ident-'next_statement'-id:0
   2 : CBC_POP
   3 : CBC_RETURN_WITH_UNDEFINED
   4 : CBC_RETURN_WITH_UNDEFINED
   5 : CBC_PUSH_IDENT ident-'next_statement'-id:0
   7 : CBC_POP
   8 : CBC_PUSH_IDENT ident-'exp'-id:1
  10 : CBC_RETURN
  11 : CBC_PUSH_IDENT ident-'next_statement'-id:0
  13 : CBC_POP
  14 : CBC_RETURN_WITH_UNDEFINED
  15 : CBC_PUSH_IDENT ident-'exp'-id:1
  17 : CBC_POP
  18 : CBC_PUSH_IDENT ident-'next_statement'-id:0
  20 : CBC_POP
  21 : CBC_RETURN_WITH_UNDEFINED

Parse successfully completed. Total byte code size: 22 bytes
